docs/source/contributing.rst

```diff
@@ -37,6 +37,13 @@ and build the documentation by changing to the ``docs/source`` directory where y
 
 The built documentation will then live in ``ucc/docs/source/_build/html``.
 
+We also use `pre-commit <https://pre-commit.com/>` to run code formatting and linting checks before each commit.
+To enable the pre-commit hooks, run
+
+.. code:: bash
+
+    pre-commit install
```

Comment:
Is there a way to make this run when `poetry install` is run? Maybe some devs wouldn't like that?

Reply:
We can look into it, but I'm actually one of those devs =). I actually just started to use https://jj-vcs.github.io/jj/latest/, which doesn't work well with pre-commit hooks currently. Instead, I'm relying on ruff's integration with vscode to format/lint as I work, and then the ci run will double check as well.